nep-12.mediawiki

```diff
@@ -0,0 +1,2130 @@
+  size: number;
+  type: string;
+  version: number;
+  attributes: TransactionAttribute[];
+  vin: any[];
+  vout: any[];
+  sys_fee: string;
+  net_fee: string;
+  scripts: TransactionScript[];
+  script: string;
+  gas: string;
+  blockhash: string;
+  confirmations: number;
+  blocktime: number;
+}
+
+interface TransactionAttribute {
+  usage: string;
+  data: string;
+}
+
+interface TransactionScript {
+  invocation: string;
+  verification: string;
+}
+</pre>
+
+
+'''Error return value'''
+
+<pre>
+interface Error {
+  type: string;
+  description: string;
+  data: string;
+}
+</pre>
+
+
+'''Example'''
+
+<pre>
+getTransaction({
+  txid: '7e049fd7c253dabf38e4156df30c78b30d49f307196aa89b99a47d2330789bf2',
+  network: 'TestNet'
+})
+.then((result: Object) => {
+  console.log('Transaction details: ' + JSON.stringigy(result));
+})
+.catch(({type: string, description: string, data: any}) => {
+  switch(type) {
+    case NO_PROVIDER:
+      console.log('No provider available.');
+      break;
+   case RPC_ERROR:
+    console.log('There was an error when broadcasting this transaction to the network.');
+    break;
+  }
+});
+</pre>
+
+
+'''Provider Request Handling'''
+
+Upon receiving this request
+* Broadcast the RPC request
+* Return the RPC response results
+
+
+Example
+<pre>
+{
+  "txid": "0x7e049fd7c253dabf38e4156df30c78b30d49f307196aa89b99a47d2330789bf2",
+  "size": 556,
+  "type": "InvocationTransaction",
+  "version": 1,
+  "attributes": [
+    {
+      "usage": "Script",
+      "data": "296ac124021a71c449a9bad320c16429b08ad6ee"
+    },
+    {
+      "usage": "Remark",
+      "data": "cbb549adec34d741"
+    }
+  ],
+  "vin": [],
+  "vout": [],
+  "sys_fee": "0",
+  "net_fee": "0",
+  "scripts": [
+    {
+      "invocation": "4072b83e8aca62c27dc36b032b895e757db00620384e26f43cd0ecc9904bff1e652dd94a03226d6dcb0b6f91104cb40be6455aa0fc3b474a8a8e5fa43ff4b10b8d40af726dc0976f15cd8a134634074c5613ab1e59979fec37b611392975c92afa11038fd9d96ddfb306df12ae200dc3c15fa17cb9530389e28f090fd8c9721c3307",
+      "verification": "53c56b6c766b00527ac46c766b51527ac4616c766b00c36121022949376faacb0c6783da8ab63548926cb3a2e8d786063a449833f927fa8853f0ac642f006c766b51c361210292a25f5f0772d73d3fb50d42bb3cb443505b15e106789d19efa4d09c5ddca756ac635f006c766b00c361210292a25f5f0772d73d3fb50d42bb3cb443505b15e106789d19efa4d09c5ddca756ac642f006c766b51c36121022949376faacb0c6783da8ab63548926cb3a2e8d786063a449833f927fa8853f0ac62040000620400516c766b52527ac46203006c766b52c3616c7566"
+    }
+  ],
+  "script": "0400e1f505147869ef9732cdf6f6d54adaa5cae3b55a9396bceb14296ac124021a71c449a9bad320c16429b08ad6ee53c1087472616e7366657267f1dfcf0051ec48ec95c8d0569e0b95075d099d84f10400e1f50514b1fdddf658ce5ff9f83e66ede2f333ecfcc0463e14296ac124021a71c449a9bad320c16429b08ad6ee53c1087472616e7366657267f1dfcf0051ec48ec95c8d0569e0b95075d099d84f1",
+  "gas": "0",
+  "blockhash": "0x4ea57fe267a392933d2b03fa733fbf1fa12c13f7e8ae2051e45465800e1a7cdb",
+  "confirmations": 9,
+  "blocktime": 1557377749
```

Comment:
This was called `time` before, not `blocktime`... we really need to ensure consistency here, for implementers.

Reply:
The return data attributes are directly from the rpc response to getrawtransaction, verbose.
https://docs.neo.org/docs/en-us/reference/rpc/latest-version/api/getrawtransaction.html

Reply:
I think we should fix these inconsistencies here
.. as far as I know, rpc was always developed on demand... this a planned interface, that can avoid transmitting these "small errors" to api users. Thats the intention of it, or no? Otherwise, why not just use rpc directly...